packages/mui-base/src/Input/Input.tsx

```diff
@@ -147,8 +147,14 @@ const Input = React.forwardRef(function Input<RootComponentType extends React.El
   const InputComponent = multiline ? slots.textarea ?? 'textarea' : slots.input ?? 'input';
   const inputProps: WithOptionalOwnerState<InputInputSlotProps> = useSlotProps({
     elementType: InputComponent,
-    getSlotProps: (otherHandlers: EventHandlers) =>
-      getInputProps({ ...otherHandlers, ...propsToForward }),
+    getSlotProps: (otherHandlers: EventHandlers) => {
+      return getInputProps({
+        ...otherHandlers,
+        ...propsToForward,
+        onKeyDown: propsToForward.onKeyDown ?? otherHandlers.onKeyDown,
+        onKeyUp: propsToForward.onKeyUp ?? otherHandlers.onKeyUp,
```

Comment:
It should be the other way round. `slotProps` should take precedence over props passed directly to the component.

This is mentioned in the [document](https://mui.com/base/getting-started/usage/#slotprops) -
> If both slotProps.root and additional props have the same keys but different values, the slotProps.root props will take precedence. This does not apply to classes or the style prop—they will be merged instead.

which was done in #33097 and which also [matches the implementation](https://github.com/mui/material-ui/blob/ec3d7dd6cbcd362736164487a9789908d5d47133/packages/mui-base/src/utils/mergeSlotProps.ts#L161-L162).

So it should be:
```js
     getInputProps({
        ...propsToForward,
        ...otherHandlers,
      }),
```

Reply:
If `getInputProps` is implemented like below, props passed in `slotProps` will always override props passed to component. 

```
     getInputProps({
        ...propsToForward,
        ...otherHandlers,
      }),
```

In that case, if component is implemented like below, `onKeyDown` won't get called because `onKeyDown` in slotProps (in this case `onKeyDown` in slotProps is `undefined`) will override `onKeyDown` passed to `component`

```
<Input onKeyDown={()=>{
console.log('here')
}} />
```

I think correct way to implement is like below, what do you think?



```
 getInputProps({
        ...propsToForward,
        ...otherHandlers,
 onKeyDown: otherHandlers.onKeyDown ?? propsToForward.onKeyDown,
        onKeyUp: otherHandlers.onKeyUp ?? propsToForward.onKeyUp,
      }),
```

Reply:
> onKeyDown won't get called because onKeyDown in slotProps (in this case onKeyDown in slotProps is undefined) will override onKeyDown passed to component

The `onKeyDown` in `slotProps` is not `undefined` in this case, but rather unavailable. In `propsToFroward`, it was declared as `undefined`, as mentioned in https://github.com/mui/material-ui/pull/37463#discussion_r1213165125. See [code](https://github.com/mui/material-ui/blob/f511b89e8b023acf0e7b221682a788a7e1a57d2c/packages/mui-base/src/Input/Input.tsx#L120-L128). It would be beneficial to create a test case to verify this.

Reply:
@ZeeshanTamboli yes, you are right. `onKeyDown` in `slotProps` will be `unavailable` not `undefined`.

>it would be beneficial to create a test case to verify this.

added test [here](https://github.com/mui/material-ui/pull/37463/files#diff-4343d3119b317ba8cd867785f6aaa4e3235148f21f82ada5644bb4c96665e363R68)